src/rgw/rgw_rados.cc

```diff
@@ -482,11 +487,9 @@ int RGWZoneGroup::read_default_id(string& default_id, bool old_format)
     /* try using default realm */
     RGWRealm realm;
     int ret = realm.init(cct, store);
-    if (ret < 0) {
-      ldout(cct, 10) << "could not read realm id: " << cpp_strerror(-ret) << dendl;
-      return -ENOENT;
+    if (ret == 0) {
+      realm_id = realm.get_id();
```

Comment:
instead of calling through `read_default_id()` in the case of ENOENT, you can call `read_id()` with the default zonegroup name - something like this?
```c++
    if (ret < 0) {
      return read_id(default_zonegroup_name, default_id);
    }
    realm_id = realm.get_id();
```